app/Resources/views/contactform/list.html.twig

```diff
@@ -34,7 +34,7 @@
     <script type="text/babel">
         var contactsUrl = '{{ path('contact_contacts') }}'
         ReactDOM.render(
-                <ContactSection url="{contactsUrl}" />,
+                <ContactSection url={contactsUrl} />,
```

Comment:
This was the biggest issue, and is a subtle ReactJS thing. If you include the comments, then literally the url becomes `{contactsUrl}`, instead of treating this like a variable and becoming `/contactform/list/contacts`. In other words, the AJAX call was going to the wrong URL (which is why it was returning HTML)